codec-http2/src/main/java/io/netty/handler/codec/http2/HpackDecoder.java

```diff
@@ -551,6 +556,14 @@ public void appendToHeaderList(CharSequence name, CharSequence value) {
             if (!exceededMaxLength) {
                 headers.add(name, value);
             }
+            // Only consider the first validation exception
+            if (validate && validationException == null) {
```

Comment:
I think I must be misunderstanding your suggestion. If it was an `else if` with the conditional above it would not validate unless `exceededMaxLength == true`, in which case it doesn't matter.

I was deliberate in letting the header be added even if validation was going to fail since there is a world where we would be interested in the results, but I could see wanting to refactor the checks such that we only add the header if it passes both checks.

Reply:
In the common (maybe all) case they will just be discarded. For the case of `exceededMaxLength == true` it made sense not to store it since it was too big, but the invalid pseudo-header case is obvious. I'll change it to not add if it fails either validation if for nothing else than to be consistent.